algoliasearch-core/src/main/java/com/algolia/search/models/indexing/PartialUpdateOperation.java

```diff
@@ -0,0 +1,53 @@
+package com.algolia.search.models.indexing;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class PartialUpdateOperation<T extends Object> {
+
+  @JsonProperty("_operation")
+  private String operation;
+
+  @JsonProperty("value")
+  private T value;
+
+  public PartialUpdateOperation(String operation, T value) {
+    this.operation = operation;
+    this.value = value;
+  }
+
+  public String getOperation() {
+    return operation;
+  }
+
+  public void setOperation(String operation) {
+    this.operation = operation;
+  }
+
+  public T getValue() {
+    return value;
+  }
+
+  public void setValue(T value) {
+    this.value = value;
+  }
+
+  public static PartialUpdateOperation<Integer> Increment(Integer value) {
```

Comment:
Maybe we should avoid capital letters for method names?
(same applies to other methods below)

We can either use lowercase letters for the methods, or replace them with inner static classes:
```java
public static class Increment extends PartialUpdateOperation<Integer> {
    public Increment(Integer value) {
      super(PartialUpdateOperationType.INCREMENT, value);
    }
  }
```

Reply:
Right, thanks a lot! I'm doing Go, Scala, C# and Java at the same time today so my casing game is a bit weird today ^^

I'll go with keeping the static functions if that's OK with you, I'm not a big fan of introducing inheritance for such a niche use-case.